packages/inter-protocol/src/psm/psmCharter.js

```diff
@@ -96,12 +96,9 @@ export const start = async zcf => {
   };
 
   const psmCharterCreatorI = M.interface('PSM Charter creatorFacet', {
-    addInstance: M.call(
-      InstanceHandleShape,
-      M.any(),
-      BrandShape,
-      BrandShape,
-    ).returns(),
+    addInstance: M.call(InstanceHandleShape, M.any())
+      .optional(BrandShape, BrandShape)
```

Comment:
TIL `.optional`. Where is it documented or implemented?

Reply:
There's [very little documentation](https://github.com/Agoric/agoric-sdk/blob/c10c36d7ccf6c85239c1dbcec9534d43b20ad00a/packages/store/src/types.js).  I dunno where the implementations are. It's tested in [test-heap-classes](https://github.com/Agoric/agoric-sdk/blob/c10c36d7ccf6c85239c1dbcec9534d43b20ad00a/packages/store/test/test-heap-classes.js#L14).

Reply:
Yeah, that documentation doesn't mention `optional`. I've noted in https://github.com/Agoric/agoric-sdk/issues/6149

Reply:
From https://github.com/Agoric/agoric-sdk/issues/6206#issuecomment-1247546215 :

FWIW, this is implemented at https://github.com/Agoric/agoric-sdk/blob/9b786284b4ff759dd050c701829df5fd0cf41749/packages/store/src/patterns/patternMatchers.js#L1537-L1543 and used several times in https://github.com/Agoric/agoric-sdk/blob/master/packages/ERTP/src/typeGuards.js

I'm actually surprised I didn't write any comments explaining this at all. I certainly agree it is needed.

Also `rest`